pemFioi/quickAlgo/python_interface.js

```diff
@@ -459,12 +459,192 @@ function LogicController(nbTestCases, maxInstructions) {
+          commaIndex += 1;
+          ret += ' ';
+        }
+        ret += "${" + snippetIndex + ':';
+        ret += proto.substring(commaIndex + 1, parenthesisCloseIndex);
+        ret += "})";
+
+        return ret;
+      }
+    }
+
+    var langTools = ace.require("ace/ext/language_tools");
+
+
+    // This array will contain all functions for which we must add autocompletion
+    var completions = [];
+
+    // we add completion on functions
+    if (this.includeBlocks && this.includeBlocks.generatedBlocks) {
+      for (var categoryIndex in this.includeBlocks.generatedBlocks) {
+        for (var funIndex in this.includeBlocks.generatedBlocks[categoryIndex]) {
+          var fun = this.includeBlocks.generatedBlocks[categoryIndex][funIndex];
+          var funInfos = this._getFunctionsInfo(fun);
+          var funProto = funInfos.proto;
+          var funHelp = funInfos.help;
+          var funSnippet = getSnippet(funProto);
+          completions.push({
+            caption: funProto,
+            snippet: funSnippet,
+            type: "snippet",
+            docHTML: "<b>" + funProto + "</b><hr></hr>" + funHelp
+          })
+        }
+      }
+      if(this._mainContext.customConstants && this._mainContext.customConstants[categoryIndex]) {
+        var constList = this._mainContext.customConstants[categoryIndex];
+        for(var iConst=0; iConst < constList.length; iConst++) {
+          var name = constList[iConst].name;
+          if(this._mainContext.strings.constant && this._mainContext.strings.constant[name]) {
+            name = this._mainContext.strings.constant[name];
+          }
+          completions.push({
+            name: name,
+            value: name,
+            meta: quickAlgoLanguageStrings[stringsLanguage].hasOwnProperty("constant")
+                    ? quickAlgoLanguageStrings[stringsLanguage].constant : "constant"
+          })
+        }
+      }
+    }
+
+    // Adding allowed consts (for, while...)
+    var allowedConsts = pythonForbiddenLists(this.includeBlocks).allowed;
+    hideHiddenWords(allowedConsts);
+
+    // This blocks are blocks which are not special but must be added
+    var toAdd = ["True", "False"];
+    for (var toAddId = 0; toAddId < toAdd.length; toAddId++) {
+      allowedConsts.push(toAdd[toAddId]);
+    }
+
+    var keywordi18n = quickAlgoLanguageStrings[stringsLanguage].hasOwnProperty("keyword")
+        ? quickAlgoLanguageStrings[stringsLanguage].keyword : "keyword";
+
+    // if we want to modify the result of certain keys
+    var specialSnippets = {
+      // list_brackets and dict_brackets are not working
+      list_brackets:
+          {
+            name: "[]",
+            value: "[]",
+            meta: keywordi18n
+          },
+      dict_brackets: {
+        name: "{}",
+        value: "{}",
+        meta: keywordi18n
+      },
+      var_assign: {
+        caption: "x =",
+        snippet: "x = $1",
+        type: "snippet",
+        meta: "variables"

```

Comment:
Use i18n there too. Also, it should be "variable" in the singular form, as you did for "mot-clé" for instance.